### DISCUSSION TITLE
如何监听到form变化，类似于antd中form表单onValuesChange这种

### DISCUSSION THREAD

**Discussion by quqiuhua** [alibaba/formily#3409] [Q&A 2.x] [answered]
背景：数据变化实时保存
代码链接： https://codesandbox.io/s/epic-snow-fw48jp?file=/App.tsx
使用了3种方法都监听不到表单值变化，求大佬指导

**Comment by janryWang** [comment, marked as answer]
createSchemaField放到渲染函数外层，createForm用useMemo包

**Reply by quqiuhua** [reply]
const form = useMemo(
    () =>
      createForm({
        effects() {
          onFormValuesChange((form) => {
            setFormValues(form.values);
          });
        }
      }),
    []
  );
  有点没太明白🤔，使用useMemo只会触发一次onFormValuesChange
